composeApp/src/commonMain/kotlin/org/ooni/probe/domain/FetchDescriptorUpdate.kt

```diff
@@ -0,0 +1,121 @@
+) {
+    private val availableUpdates = MutableStateFlow(DescriptorUpdatesStatus())
+
+    suspend fun invoke(
+        descriptors: List<InstalledTestDescriptorModel>,
+    ): MutableMap<ResultStatus, MutableList<Result<InstalledTestDescriptorModel?, MkException>>> {
+        availableUpdates.update { _ ->
+            DescriptorUpdatesStatus(
+                refreshType = UpdateStatusType.FetchingUpdates,
+            )
+        }
+        val response = coroutineScope {
+            descriptors.map { descriptor ->
+                async {
+                    Pair(descriptor, fetchDescriptor(descriptor.id.value.toString()))
+                }
+            }.awaitAll()
+        }
+        val autoUpdateItems = mutableSetOf<InstalledTestDescriptorModel>()
+        val resultsMap = mutableMapOf<ResultStatus, MutableList<Result<InstalledTestDescriptorModel?, MkException>>>()
+        ResultStatus.entries.forEach { resultsMap[it] = mutableListOf() }
+        response.forEach { (descriptor, result) ->
+            val status: ResultStatus = if (descriptor.autoUpdate) {
+                var res: ResultStatus = ResultStatus.NoUpdates
+                result.map { updatedDescriptor ->
+                    updatedDescriptor?.let {
+                        if (descriptor.shouldUpdate(it)) {
+                            autoUpdateItems.add(it.copy(autoUpdate = descriptor.autoUpdate))
+                            res = ResultStatus.AutoUpdated
+                        }
+                    }
+                }
+                res
+            } else {
+                var res: ResultStatus = ResultStatus.NoUpdates
+                result.map { updatedDescriptor ->
+                    updatedDescriptor?.let {
+                        if (descriptor.shouldUpdate(updatedDescriptor)) {
+                            res = ResultStatus.UpdatesAvailable
+                        }
+                    }
+                }
+                res
+            }
+            resultsMap[status]?.add(result.map { it?.copy(autoUpdate = descriptor.autoUpdate) })
+        }
+
+        createOrUpdateTestDescriptors(autoUpdateItems)
+        val updatesAvailable: List<InstalledTestDescriptorModel> = resultsMap[ResultStatus.UpdatesAvailable]?.mapNotNull { result ->
+            result.get()
+        }.orEmpty()
+        val autoUpdated: List<InstalledTestDescriptorModel> = resultsMap[ResultStatus.AutoUpdated]?.mapNotNull { result ->
+            result.get()
+        }.orEmpty()
+        availableUpdates.update { _ ->
+            DescriptorUpdatesStatus(
+                availableUpdates = updatesAvailable.toSet(),
+                autoUpdated = autoUpdated.toSet(),
+                refreshType = if (updatesAvailable.isNotEmpty()) UpdateStatusType.ReviewLink else UpdateStatusType.None,
+            )
+        }
+        return resultsMap
+    }
+
+    suspend operator fun invoke() {
+        listInstalledTestDescriptors.invoke().first().let { items ->
```

Comment:
`listInstalledTestDescriptors.invoke()` is the same as `listInstalledTestDescriptors()`